exchange-app/src/app/services/state.service.ts

```diff
@@ -1,63 +1,61 @@
 import { Injectable } from '@angular/core';
 import { BehaviorSubject, Observable } from 'rxjs';
-import exchangerate from '../interfaces/exchangerate';
-import {currencyState, valueState} from '../interfaces/state';
+import Exchangerate from '../interfaces/Exchangerate';
+import { CurrencyState, ValueState } from '../interfaces/State';
 
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class StateService {
-  private currencyState = new BehaviorSubject<currencyState>({}) 
-  private valueState = new BehaviorSubject<valueState>({})
-  private exchangeRateState = new BehaviorSubject<exchangerate>({})
-  
-  getCurrencyState(): Observable<currencyState>{
-    return this.currencyState.asObservable()
+  private currencyState = new BehaviorSubject<CurrencyState>({});
+  private valueState = new BehaviorSubject<ValueState>({});
+  private exchangeRateState = new BehaviorSubject<Exchangerate>({});
+
+  getCurrencyState(): Observable<CurrencyState> {
+    return this.currencyState.asObservable();
   }
 
-  setCurrencyState(from?:string, to?:string): void{
-    if(from===to){
-      this.swapCurrency()
-    } else{
+  setCurrencyState(from?: string, to?: string): void {
+    if (from === to) {
+      this.swapCurrency();
+    } else {
       this.currencyState.next({
         from: from,
-        to: to
-      })
+        to: to,
+      });
 
-      this.setValueState(NaN,NaN)
+      this.setValueState(NaN, NaN);
     }
-  }  
+  }
 
-  swapCurrency(): void{
+  swapCurrency(): void {
     this.currencyState.next({
       from: this.currencyState.value.to,
-      to: this.currencyState.value.from
-    })
+      to: this.currencyState.value.from,
+    });
 
     this.valueState.next({
       fromValue: this.valueState.value.toValue,
-      toValue: this.valueState.value.fromValue
-    })
+      toValue: this.valueState.value.fromValue,
+    });
   }
 
-  getValueState(): Observable<valueState>{
-    return this.valueState.asObservable()
+  getValueState(): Observable<ValueState> {
+    return this.valueState.asObservable();
```

Comment:
`public this.valueState$: Observable<ValueState> = this.valueState.asObservable()`